sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -161,6 +162,8 @@ def __init__(self, name, init_config, instances):
             ttl=60 * 60 * 24,
         )
 
+        self.check_initializations.append(self.set_resolved_hostname_metadata)
```

Comment:
nit: I don't think this even needs to be a check initialization. The fact that the `resolved_hostname` is a cached property means it will only run once anyway.

Reply:
This must be included in a check initialization because `set_metadata` can only be called successfully after init has completed, once the Agent has set the `check_id`. Adding this to check_initializations ensures that this gets called the first time we call `run`. [[slack thread for more context](https://dd.slack.com/archives/C9WBCH0AF/p1678128632653349)]  <img width="632" alt="Screen Shot 2023-03-07 at 1 08 27 PM" src="https://user-images.githubusercontent.com/6165248/223553472-a216b0d5-ec10-42e3-8292-8c23426d2280.png">

Reply:
Ah ok, then please disregard my nit :)